dot/sync/syncer_integration_test.go

```diff
@@ -125,7 +117,7 @@ func newTestSyncer(t *testing.T) *Service {
 	}).AnyTimes()
 
 	cfg.FinalityGadget = mockFinalityGadget
-	cfg.Network = newMockNetwork()
+	cfg.Network = NewMockNetwork(ctrl)
```

Comment:
You can probably leave it as nil if there is no expectations at all.

```suggestion
```

Reply:
NewService does a nil check on cfg.Network here

Reply:
Ugh ugly, but out of scope, fine!!! 😄